### PR TITLE
Add BBE tests for Type Check Expression

### PR DESCRIPTION
**Pull request by warunalakshitha**
## Purpose
$Subject

### CODE REVIEW

**Comment by MaryamZi** [comment]
Is this BBE listed on bio? 

We have https://ballerina.io/learn/by-example/type-guard.html for the `is` check.

**Comment by SupunS** [comment]
we can do 'type test' for any expression. eg: `foo() is T`. 'implicit type narrowing' will not always applies to type test expressions.
Type guard is a special case of type test, where the expr is a variable-ref.

May be we should update the example stating that?

File: examples/type-check-expression/type_check_expression.description [hunk 1]
```diff
@@ -1,2 +1,2 @@
-// Type check expression can be used to assert the runtime type of a value in Ballerina.
+// Type check expression can be used to assert the runtime type of a value or any expression in Ballerina.
```
Comment (SupunS) [inline examples/type-check-expression/type_check_expression.description]: The previous statement sound right to me. It always checks the type of a value at runtime.

Being able to give any expression during compile time is different to that IMO.

File: examples/type-check-expression/tests/type_check_expression_test.bal [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+public function mockPrint(any|error... s) {
+    foreach var elem in s {
+        outputs[counter] = elem;
+        counter += 1;
+    }
+}
+
+@test:Config
+function testFunc() {
+    // Invoking the main function
```
Comment (Shan1024) [inline examples/type-check-expression/tests/type_check_expression_test.bal]:
```suggestion
    // Invoking the main function.
```

Comment (warunalakshitha) [reply examples/type-check-expression/tests/type_check_expression_test.bal]: Fixed

File: examples/type-check-expression/type_check_expression.description (hunk 1 above)
Comment (warunalakshitha) [reply examples/type-check-expression/type_check_expression.description]: Revert to previous statement.

File: examples/type-check-expression/type_check_expression.bal [hunk 3]
```diff
@@ -24,7 +24,7 @@ public function main() {
     if (x is Student) {
         io:println("Alex is a student");
     } else {
-        io:println("Alex is a student");
+        io:println("Alex is not a student");
     }
 
     // Type of `x` is `Student`. But it is structurally equivalent to `Person`.
```
Comment (praneesha) [inline examples/type-check-expression/type_check_expression.bal]: Type of `x` is `Student`. However, it is structurally equivalent to `Person`.

File: examples/type-check-expression/type_check_expression.bal [hunk 4]
```diff
@@ -41,6 +41,12 @@ public function main() {
     } else {
         io:println("Alex is not a vehicle");
     }
+
+    // Type check expression can be used for any expression.
```
Comment (praneesha) [inline examples/type-check-expression/type_check_expression.bal]: Type `check` can be used for any expression.

**Pull request merged**